### PR TITLE
Prepare LLR framework to run on 2017 and 2018 at the same time

### PR DESCRIPTION
**Pull request by francescobrivio**
We can now use this branch (102X_HH) to run on both 2017 and 2018 samples including deepFlavor and deepTau. There is a new variable called "YEAR" and set in analyzer.py that is used to customize all the rest of the framework to select the recommended recipes for the different years.
In particulare the customized sections are:
- JEC uncertainties
- JET ID
- TES
- MET filters
- lumi mask
- HLT trigger list
- Global Tag
- PostRecoSequence for Electrons
- VBF: removed unused trig match and fixed some filters for the used one

@camendola and @dzuolo can you think of something else that should be adapted to run on 2017 and 2018 at the same time?

### CODE REVIEW

**Comment by dzuolo** [comment]
@francescobrivio Did you check the ECAL fixEE in the producer?

**Comment by francescobrivio** [comment]
thanks @dzuolo I missed that!
Now it is included: actually it turns out that for CMSSW_10_2_X (X>=7) this fix is included in CMSSW by default:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/MissingETUncertaintyPrescription#Instructions_for_10_2_X_X_7
So I commented it in the producer and removed the instruction from the README

**Comment by francescobrivio** [comment]
Updated TES values for DeepTau according to:
https://indico.cern.ch/event/864131/contributions/3644021/attachments/1946837/3230164/Izaak_TauPOG_TauES_20191118.pdf

**Comment by camendola** [comment]
Ciao @francescobrivio and @dzuolo , sorry for coming on this so late, thanks for these implementations! Naive question: what about 2016? Do we need to implement these things also in the separate branch for 2016 or should we add a 2016 option here?

**Comment by francescobrivio** [comment]
ciao @camendola from what I understood 2016 data would be included only at "datacard level" so no need to reprocess them. But if things have changed we need to add that too...

**Comment by giacomoortona** [comment]
Hi @francescobrivio @camendola @dzuolo 
Since this is a legacy measurement, I think we need to at least have the capability to reprocess 2016 data.

**Comment by francescobrivio** [comment]
hi @giacomoortona ok then, we will add the 2016 part as well!
@camendola if you don't see any problem, this PR can be merged in the meanwhile

File: NtupleProducer/python/HiggsTauTauProducer_102X.py [hunk 1]
```diff
@@ -340,29 +340,51 @@
...
+# 2018 data - MVAoldDM2017v2
+#if YEAR == 2018:
+#    NomTESUnc1Pr      = cms.double(1.1)  # in percent, up/down uncertainty of TES
+#    NomTESUnc1PrPi0   = cms.double(0.9)  # in percent, up/down uncertainty of TES
+#    NomTESUnc3Pr      = cms.double(0.8)  # in percent, up/down uncertainty of TES
+#    NomTESUnc3PrPi0   = cms.double(1.0)  # in percent, up/down uncertainty of TES -- Missing for 2018, kept the same of 2017
+#    NomTESCor1Pr      = cms.double(-1.3) # DecayMode==0
+#    NomTESCor1PrPi0   = cms.double(-0.5) # DecayMode==1
+#    NomTESCor3Pr      = cms.double(-1.2) # DecayMode==10
+#    NomTESCor3PrPi0   = cms.double(-0.1) # DecayMode==11 -- Missing for 2018, kept the same of 2017
```
Comment (camendola) [inline NtupleProducer/python/HiggsTauTauProducer_102X.py:366]: Should we maybe get rid of commented lines like these ones?

Comment (francescobrivio) [reply NtupleProducer/python/HiggsTauTauProducer_102X.py:366]: these are the TES values for the old tauID (MVAv2) so I prefer to keep them commented here

**Pull request merged**